js/app.js

```diff
@@ -0,0 +1,306 @@
+console.log(getReversedString('Arrays')); // 'syarrA'
+
+// 5
+function getReversedString(str) {
+  const lowercaseNoSpaces = str.replace(/\s+/g, '').toLowerCase('');
+  const splittedString = lowercaseNoSpaces.split('');
+  const reversedArray = splittedString.reverse('');
+  const reversedString = reversedArray.join('');
+  return reversedString;
+}
+
+function isPalindrome(str) {
+  const reversedStr = getReversedString(str);
+  const normalizedStr = str.replace(/\s+/g, '').toLowerCase('');
+  return normalizedStr === reversedStr;
+}
+
+console.log(isPalindrome('Kayak'));
+console.log(isPalindrome('Racecar'));
+console.log(isPalindrome('Was it a cat I saw'));
+console.log(isPalindrome('Hello!'));
+
+// 6
+function countLetters(text) {
+  const letterCounts = {};
+  const lowercaseNoSpaces = text.replace(/\s+/g, '').toLowerCase();
+
+  for (let letter of lowercaseNoSpaces) {
+    if (letter >= 'a' && letter <= 'z') {
+      if (letterCounts[letter]) {
+        letterCounts[letter] += 1;
+      } else {
+        letterCounts[letter] = 1;
+      }
+    }
+  }
+  return letterCounts;
+}
+
+const lettersObject = countLetters('The quick brown fox jumps over the lazy dog');
+
+console.log(lettersObject);
+
+// 7
+function getYoungestPerson(people) {
+  let youngest = people[0];
+  for (let person of people) {
+    if (person.age < youngest.age) {
+      youngest = person;
+    }
+  }
+  return youngest;
+}
+
+function getOldestPerson(people) {
+  let oldest = people[0];
+  for (let person of people) {
+    if (person.age > oldest.age) {
+      oldest = person;
+    }
+  }
+  return oldest;
+}
+
+function getAgeDifference(people) {
+  const youngest = getYoungestPerson(people);
+  const oldest = getOldestPerson(people);
+  return oldest.age - youngest.age;
+}
+
+const peopleArray = [
+  {
+    name: 'Adam',
+    age: 20
+  },
+  {
+    name: 'Amanda',
+    age: 5
+  },
+  {
+    name: 'John',
+    age: 75
+  },
+  {
+    name: 'Dave',
+    age: 15
+  }
+]
+
+console.log(getAgeDifference(peopleArray)); // 70
+
+// 8
+function positiveSum(arr) {
+  let sum = 0;
+
+  for (let number of arr) {
+    if (number > 0) {
+      const positiveNumber = number;
+      const newSum = sum + positiveNumber;
+      sum = newSum;
+    }
+  }
+
+  return sum
+}
+
+// 9
+function squareSum(numbers) {
+  let sum = 0;
+
+  for (let number of numbers) {
+
+    const square = number * number;
+    sum = sum + square;
+
+  }
+
+  return sum
+}
+
+// 11
+function nameSplit(name) {
+  return name.split(' '); // Split the name by the space and return the array
+}
+
+console.log(nameSplit("Sam Harris"));
+
+function abbrevName(name) {
+  const nameArray = nameSplit(name);
+
+  const firstInitial = nameArray[0][0].toUpperCase();
+  const secondInitial = nameArray[1][0].toUpperCase();
+
+  return `${firstInitial}.${secondInitial}`
+}
+
+// 12
+function countSheeps(sheep) {
+  let count = 0;
+
+  for (let i = 0; i < sheep.length; i++) {
+    if (sheep[i] === true) {
```

Comment:
```suggestion
    if (sheep[i]) {
```
`sheep[i]` is already true or false, no need to compare it to `true`